core/src/lib.rs

```diff
@@ -139,6 +138,12 @@ pub enum AttributeBitFlags {
     UniqueValue   = 1 << 3,
 }
 
+#[derive(Clone,Debug,Eq,Hash,Ord,PartialOrd,PartialEq)]
+pub enum Unique {
```

Comment:
Can we call this `AttributeUniqueness` or something more specific? This is in `core`.